File: src/main/java/proto/games/matchmaker/configuration/properties/DefaultProperties.java
package proto.games.matchmaker.configuration.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Data
@Configuration
@ConfigurationProperties(prefix = "default")
public class DefaultProperties {
    private TeamProperties team;

    @Data
    public static class TeamProperties {
        private int size;
    }
}

File: src/main/kotlin/proto/games/matchmaker/configuration/properties/DefaultProperties.kt
package proto.games.matchmaker.configuration.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "default")
class DefaultProperties {
    var team: TeamProperties = TeamProperties()

    class TeamProperties {
         var size: Int = 0
    }
}